test/gtest/ucs/test_datatype.cc

```diff
@@ -1356,3 +1359,358 @@ UCS_MT_TEST_F(test_datatype_ptr_map_safe, safe_put, num_threads)
+            start = seg->end + 1;
+        }
+    }
+
+    void compare_point_value(ucs_piecewise_func_t &piecewise_func,
+                             const ucs_linear_func_t &linear_func, size_t point)
+    {
+        ASSERT_EQ(ucs_piecewise_func_apply(&piecewise_func, point),
+                  ucs_linear_func_apply(linear_func, point))
+                  << "point is " << point;
+    }
+
+    size_t get_num_segments(const ucs_piecewise_func_t &func)
+    {
+        size_t num_ranges = 0;
+        ucs_piecewise_segment_t *seg;
+
+        ucs_list_for_each(seg, &func.head, list) {
+            ++num_ranges;
+        }
+        return num_ranges;
+    }
+
+    void check_signle_segment_func(const ucs_piecewise_func_t &func,
+                                   const ucs_linear_func_t &segment_func)
+    {
+        ASSERT_EQ(get_num_segments(func), 1);
+        ASSERT_TRUE(ucs_linear_func_is_equal(func.head_segment.func,
+                                             segment_func, 0));
+    }
+
+    void check_funcs_sum(ucs_piecewise_func_t &result,
+                         std::vector<ucs_piecewise_func_t> &funcs)
+    {
+        size_t seg_start = 0;
+        std::set<size_t> points;
+        ucs_piecewise_segment_t *seg;
+
+        for (auto &func : funcs) {
+            ucs_list_for_each(seg, &func.head, list) {
+                /* Test start, end and middle of the each segment */
+                points.insert(seg_start);
+                points.insert(seg->end);
+                points.insert(seg_start + (seg->end - seg_start) / 2);
+
+                seg_start = seg->end + 1;
+            }
+        }
+
+        for (size_t point : points) {
+            double expected = 0;
+            for (auto &func : funcs) {
+                expected += ucs_piecewise_func_apply(&func, point);
+            }
+
+            double actual = ucs_piecewise_func_apply(&result, point);
+
+            if (actual != expected) {
+                for (const auto &func : funcs) {
+                    print_piecewise_func(func);
+                }
+            }
+            ASSERT_EQ(actual, expected) << "point is " << point;
+        }
+    }
+
+    size_t get_random_value(size_t min, size_t max)
+    {
+        if (max == min) {
+            return min;
+        }
+        return min + (ucs::rand() % (max - min));
+    }
+
+    segment get_random_segment(size_t max_end = SIZE_MAX)
+    {
+        size_t start = get_random_value(0, max_end);
+        size_t end   = get_random_value(start, max_end);
+        auto func    = ucs_linear_func_make(get_random_value(0, 10),
+                                            get_random_value(0, 10));
+
+        return {start, end, func};
+    }
+
+    void piecewise_func_add_segment(ucs_piecewise_func_t *func, size_t start,
+                                    size_t end, ucs_linear_func_t trend)
+    {
+        ucs_status_t status;
+        status = ucs_piecewise_func_add_segment(func, start, end, trend);
```

Comment:
`ASSERT_UCS_OK(ucs_piecewise_func_add_segment(func, start, end, trend))` for all occurences?

Reply:
This PR contains some outdated code state. This is done in this PR (https://github.com/openucx/ucx/pull/9569) which should be merged first.